hack4impact-calpoly/ecologistics-shared-calendar
# 91 admin events page backend

## Developer: Belal Elshenety

Closes #91 

### Pull Request Summary

Populated adminEvents page from the backend. Approve, deny, delete buttons works with the endpoints in eventRoutes.ts
### Modifications

EventRequestTable.tsx: updated with new schema, and Date and time to be identical to figma
eventSchema.ts: added deniedReason to schema, which will contain our decline message
middleware.ts: eventdetails.tsx is public to show eventDetails without logging in, and adminEvent is only limited to admins
eventRoutes.ts: added PATCH method to be used when denying and approving events
adminEvents.tsx: populated with events from backend and removed hard coded events
### Testing Considerations

- Make new event
- Approve and check that it moves to approved events
- Deny, check it moved to Declined events, and check that decline message populates in the backend
- Delete events and check they don't exist on the page or backend
- Check that only admins can access adminEvents page
### Pull Request Checklist


### Screenshots/Screencast
<img width="1511" alt="Screenshot 2024-05-14 at 7 24 14 PM" src="https://github.com/hack4impact-calpoly/ecologistics-shared-calendar/assets/113317327/cacf40e5-4300-4fbc-a766-7d6f98231323">
